strawberryfields/api/connection.py

```diff
@@ -154,6 +154,8 @@ def create_job(self, target: str, program: Program, run_options: dict = None) ->
 
         circuit = bb.serialize()
 
+        self.log.debug("Submitting job\n%s", circuit)
```

Comment:
Would be good to keep the level consistent here and on line 164: `self.log.info("Job %s was successfully submitted.", job_id)`. Either both could be `DEBUG` or both could be `INFO`.

One thing I was wondering about: for errors, how should we handle logging? Would it make sense to have the same message when instantiating the error and also log that at the same time?

Reply:
Good question! Errors should definitely be logged, the generated log files should have a running commentary of what happens during program execution.

One way of doing this could be using [`logger.exception`](https://docs.python.org/3/library/logging.html#logging.Logger.exception). 

This article seems to provide a nice overview of potential logging/exception patterns: https://www.loggly.com/blog/exceptional-logging-of-exceptions-in-python/

Reply:
I originally had this as `info`, but I'm not so sure they both belong at the same level. From the Python docs:

Level | Description
-- | --
`DEBUG` | Designates fine-grained informational events that are most useful to debug an application.
`INFO` | Designates informational messages that highlight the progress of the application at coarse-grained level.



* Job submission provides information about the progress of the job submission at a very course-grained level. Seems particularly well suited, given that `info` is the default level, and will be displayed on the output by default.

* Logging the submitted circuit is instead more fine-grained debugging data. In most cases, users won't want this displayed as 'info', since the submitted circuit should be evident from their Python code (or determined via `prog.compile()`). However, this will be useful information when debugging.

Reply:
Sounds good! Happy if these are kept as they are right now :)